incubator/group/keeper_test.go

```diff
@@ -194,18 +198,163 @@ func TestCreateGroupAccount(t *testing.T) {
+	require.NoError(t, err)
+
+	policy = group.ThresholdDecisionPolicy{
+		Threshold: sdk.NewDec(math.MaxInt64),
+		Timout:    types.Duration{Seconds: 1},
+	}
+	bigThresholdAddr, err := k.CreateGroupAccount(ctx, []byte("valid--admin-address"), myGroupID, policy, "test")
+	require.NoError(t, err)
+
+	specs := map[string]struct {
+		srcAccount   sdk.AccAddress
+		srcProposers []sdk.AccAddress
+		srcMsgs      []sdk.Msg
+		srcComment   string
+		expErr       bool
+	}{
+		"all good with minimal fields set": {
+			srcAccount:   accountAddr,
+			srcProposers: []sdk.AccAddress{[]byte("valid-member-address")},
+		},
+		"all good with good msg payload": {
+			srcAccount:   accountAddr,
+			srcProposers: []sdk.AccAddress{[]byte("valid-member-address")},
+			srcMsgs:      []sdk.Msg{&testdata.MyAppProposalPayloadMsgA{}, &testdata.MyAppProposalPayloadMsgB{}},
+		},
+		"invalid payload should be rejected": {
+			srcAccount:   accountAddr,
+			srcProposers: []sdk.AccAddress{[]byte("valid-member-address")},
+			srcMsgs:      []sdk.Msg{testdata.MyAppProposalPayloadMsgA{}},
+			srcComment:   "payload not a pointer",
+			expErr:       true,
+		},
+		"comment too long": {
+			srcAccount:   accountAddr,
+			srcComment:   strings.Repeat("a", 256),
+			srcProposers: []sdk.AccAddress{[]byte("valid-member-address")},
+			expErr:       true,
+		},
+		"group account required": {
+			srcComment:   "test",
+			srcProposers: []sdk.AccAddress{[]byte("valid-member-address")},
+			expErr:       true,
+		},
+		"existing group account required": {
+			srcAccount:   []byte("non-existing-account"),
+			srcProposers: []sdk.AccAddress{[]byte("valid-member-address")},
+			expErr:       true,
+		},
+		"impossible case: decision policy threshold > total group weight": {
+			srcAccount:   bigThresholdAddr,
+			srcProposers: []sdk.AccAddress{[]byte("valid-member-address")},
+			expErr:       true,
+		},
+		"only group members can create a proposal": {
+			srcAccount:   accountAddr,
+			srcProposers: []sdk.AccAddress{[]byte("non--member-address")},
+			expErr:       true,
+		},
+		"all proposers must be in group": {
+			srcAccount:   accountAddr,
+			srcProposers: []sdk.AccAddress{[]byte("valid-member-address"), []byte("non--member-address")},
+			expErr:       true,
+		},
+		"proposers must not be nil": {
+			srcAccount:   accountAddr,
+			srcProposers: []sdk.AccAddress{[]byte("valid-member-address"), nil},
+			expErr:       true,
+		},
+		"admin that is not a group member can not create proposal": {
+			srcAccount:   accountAddr,
+			srcComment:   "test",
+			srcProposers: []sdk.AccAddress{[]byte("valid--admin-address")},
+			expErr:       true,
+		},
+	}
+	for msg, spec := range specs {
+		t.Run(msg, func(t *testing.T) {
+			id, err := k.CreateProposal(ctx, spec.srcAccount, spec.srcComment, spec.srcProposers, spec.srcMsgs)
+			if spec.expErr {
+				require.Error(t, err)
+				return
+			}
+			require.NoError(t, err)
+
+			// then all data persisted
+			loaded, err := k.GetProposal(ctx, id)
```

Comment:
`loaded` is bit ambiguous
```suggestion
			proposal, err := k.GetProposal(ctx, id)
```